### DISCUSSION TITLE
TypeScript Error

### DISCUSSION THREAD

**Discussion by andipyk** [gilbarbara/react-inlinesvg#184] [Q&A] [answered]
Hello there,
I have been using your library for ages, but all of the sudden, I am getting this weird TS error:

`TS2786: 'SVG' cannot be used as a JSX component. Its instance type 'InlineSVG' is not a valid JSX element.`

Do you, please, have any idea what might be causing the error?

same issue with but case closed
https://github.com/gilbarbara/react-inlinesvg/issues/181

>   "dependencies": {
>     "@formatjs/intl-pluralrules": "^4.0.28",
>     "@formatjs/intl-relativetimeformat": "^9.1.7",
>     "@fortawesome/fontawesome-free": "^5.15.3",
>     "@manaflair/redux-batch": "^1.0.0",
>     "@popperjs/core": "~2.10.1",
>     "@reduxjs/toolkit": "^1.6.0",
>     "animate.css": "^4.1.1",
>     "apexcharts": "^3.27.1",
>     "axios": "^0.21.1",
>     "axios-mock-adapter": "^1.19.0",
>     "bootstrap": "^5.1.3",
>     "bootstrap-icons": "^1.7.2",
>     "chart.js": "^3.3.2",
>     "clsx": "^1.1.1",
>     "del": "^6.0.0",
>     "formik": "^2.2.9",
>     "line-awesome": "^1.3.0",
>     "lint-staged": "^11.0.0",
>     "mini-css-extract-plugin": "1.6.2",
>     "nouislider": "^15.2.0",
>     "object-path": "^0.11.5",
>     "path": "^0.12.7",
>     "prettier": "^2.3.2",
>     "prism-react-renderer": "^1.2.1",
>     "prism-themes": "^1.7.0",
>     "prismjs": "^1.23.0",
>     "qs": "6.10.3",
>     "react": "^17.0.2",
>     "react-apexcharts": "^1.3.9",
>     "react-bootstrap": "^2.1.0",
>     "react-copy-to-clipboard": "^5.0.3",
>     "react-dom": "^17.0.2",
>     "react-inlinesvg": "^2.3.0",
>     "react-intl": "^5.20.4",
>     "react-is": "^17.0.2",
>     "react-query": "^3.34.12",
>     "react-redux": "^7.2.4",
>     "react-router-dom": "6.2.1",
>     "react-scripts": "4.0.3",
>     "react-table": "^7.7.0",
>     "react-topbar-progress-indicator": "^4.1.0",
>     "redux": "^4.1.0",
>     "redux-persist": "^6.0.0",
>     "redux-saga": "^1.1.3",
>     "rtlcss-webpack-plugin": "^4.0.6",
>     "sass": "1.32.1",
>     "socicon": "^3.0.5",
>     "typescript": "^4.1.2",
>     "web-vitals": "^1.0.1",
>     "yup": "^0.32.9"
>   },
>   "devDependencies": {
>     "@testing-library/jest-dom": "^5.11.4",
>     "@testing-library/react": "^11.1.0",
>     "@testing-library/user-event": "^12.1.10",
>     "@types/bootstrap": "^5.0.16",
>     "@types/chart.js": "2.9.32",
>     "@types/jest": "^26.0.15",
>     "@types/node": "^12.0.0",
>     "@types/nouislider": "15.0.0",
>     "@types/object-path": "0.11.0",
>     "@types/prismjs": "1.16.5",
>     "@types/qs": "6.9.7",
>     "@types/react": "^17.0.0",
>     "@types/react-copy-to-clipboard": "5.0.0",
>     "@types/react-dom": "^17.0.0",
>     "@types/react-redux": "7.1.16",
>     "@types/react-table": "^7.7.9",
>     "react-query-devtools": "2.6.3",
>     "v": "^0.3.0",
>     "webpack-cli": "^4.8.0"
>   }
>

**Comment by gilbarbara** [comment]
@andipyk 

As you mentioned, it was working and now is throwing TS errors, so logically isn't something with the library, right?
You are using old dependencies, so they might be the culprit...

**Comment by Themodem** [comment, marked as answer]
```diff
- "@types/react": "^17.0.0",
+ "@types/react": "^18.0.8",
```

In my case bumping `@types/react` to `^18.0` was enough to resolve this issue
